notebooks/corridor/corridor_bucharest.qmd

```diff
@@ -413,7 +423,107 @@ st_write(
         st_union(corridor_edge_1),
         st_union(corridor_edge_2)
     ),
-    dsn = sprintf("data/generated/corridor_%s.gpkg", river_name),
+    dsn = sprintf("%s/corridor_%s.gpkg", output_dir, river_name),
     append=FALSE,
 )
 ```
+
+### Define some functions to calculate a different weight
+
+In this approach, the distance from the initial corridor edges is incorporated into the edge length as a "weight" before calculating the shortest path. This added cost encourages the path to stay as close as possible to the initial corridor. When comparing two edges of the same length, the edge closer to the initial corridor is preferred, as its nodes have smaller distances to the corridor. Conversely, when comparing edges with equal distances to the corridor, the one with the shorter length is selected. Therefore, the "weight" of an edge is determined by its length plus the sum of its distances from the corridor.
+
+```{r}
+# add id to nodes
+add_node_id <- function(net){
+    net |>
+        activate("nodes") |>
+        mutate(node_id = row_number())
+}
+
+# add node distance to nodes
+add_node_dist <- function(net, buffer){
+    nodes <- getNodes(net)
+    distances <- st_distance(nodes, buffer, which = "Euclidean")
+    net |>
+        activate("nodes") |>
+        mutate(node_dist = distances)
+}
+
+# add edge_length to edges
+add_edge_length <- function(net){
+    net |>
+        activate("edges") |>
+        mutate(edge_length = edge_length())
+}
+
+# add sum of distances to each edge
```

Comment:
This is a nice way to get some sort of distance between an edge and the region, by using the sum of the distances to the start- and end- nodes of the edge. 

Maybe even simpler, could one use the distance computed directly from an edge to the corridor? I have tried to run the following:
```
edges <- network |> activate("edges") |> st_geometry()
dists <- st_distance(edges, corridor_initial, which = "Euclidean") 
```
which **I think** returns the shortest distance between each edge and the corridor. Do you think this could work?

Reply:
Yes, you are right. in this case, `st_distance` selects the minimum value. I was thinking about a case like below, which edge (1 or 2) is selected considering only one node (the nearest to the initial corridor)? I did a test for this notebook. The final results were the same. So, I replaced the sum of the distances of nodes with your suggestions. It makes the code simpler. 
![network_dist](https://github.com/user-attachments/assets/8551d0ec-705f-46f4-b145-d9ea6803c013)